Analytical_Engine/Analytical_Engine.csproj

```diff
@@ -95,12 +95,12 @@
     <Compile Include="Query\ExternalPolycurve.cs" />
     <Compile Include="Query\Geometry.cs" />
     <Compile Include="Query\Graph\Adjacency.cs" />
+    <Compile Include="Query\Graph\DistinctEntities.cs" />
     <Compile Include="Query\Graph\EntityNeighbourhood.cs" />
```

Comment:
Need to rename this CompareEntities

Reply:
Actually I think the name could stay as it is (DistinctEntities) because you do not only compare the entities, but the end result of the function is getting the unique entities (output is a Dictionary).

So either rename it to UniqueEntities() or leave it as DistinctEntities(), but I would avoid the name "compare" as not fully descriptive of the function.